### ISSUE TITLE
Azure OpenAI API Support

### ISSUE THREAD

**Issue by crypticatul** [rashadphz/farfalle#46]
Hi, 

I want to utilize my Azure OpenAI API for this, can you add and option for the same? GPT-3.5-Turbo, and GPT-4o will be enough as an option. I am deploying on Vercel.

Thanks.

**Comment by rashadphz** [comment]
Hey, I added support for azure! Let me know if you have any trouble setting it up!

**Comment by crypticatul** [comment]
Hi, thanks for adding the support for Azure. I just have a few silly queries;
In env vars
- I have deployed gpt-35-turbo-16k with name "Better" so what should I use in AZURE_DEPLOYMENT_NAME
- What we need to put in AZURE_API_BASE? is it region name or Azure Endpoint Name? I have deployed this in eastus
- For Azure the OPENAI_MODE=azure, right?

I'm using the below env vars and still getting error - OPENAI_API_KEY environment variable not found

<img width="744" alt="image" src="https://github.com/rashadphz/farfalle/assets/88734298/d6785f81-dbd8-4fee-9482-dd69545d2b97">
